src/lib/ui/toast.ts

```diff
@@ -1,32 +1,161 @@
-export function show_toast(baheth_link, type: "video" | "playlist") {
+import { get_settings } from "../storage";
+
+// Helper function to get extension URL that works with TypeScript
+function getExtensionURL(path: string): string {
+  // Access the global namespace using Window interface
+  const win = window as any;
+  
+  // Chrome, Edge, Opera
+  if (win.chrome?.runtime?.getURL) {
+    return win.chrome.runtime.getURL(path);
+  }
+  // Firefox
+  else if (win.browser?.runtime?.getURL) {
+    return win.browser.runtime.getURL(path);
+  }
+  // Fallback
+  return path;
+}
+
+export async function show_toast(baheth_link, type: "video" | "playlist", timeout = 0, isLoading = false) {
+  // Get settings for timeout if not explicitly provided
+  if (timeout === 0) {
+    const settings = await get_settings();
+    timeout = settings.notification_timeout || 6000; // Default to 6 seconds if not set
+  }
+  
+  // Detect dark mode
+  const prefersDarkMode = window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches;
+  
+  // Remove existing toasts before showing a new one
+  if (!isLoading) {
+    delete_all_toasts();
+  }
+  
   // create toast element
   const toast = document.createElement("div");
   toast.classList.add("baheth-toast");
-  toast.innerHTML = `
-    <p class="toast-title">${
-      type === "video" ? "هذا المقطع متاح" : "قائمة التشغيل هذه متاحة"
-    } على باحث! 🔍</p>
-    <p class="toast-description">اضغط للمشاهدة عبر باحث.</p>
-    <button class="close">تجاهل</button>
-  `;
+  if (prefersDarkMode) {
+    toast.classList.add("dark-mode");
+  }
+  
+  if (isLoading) {
+    toast.classList.add("loading-toast");
+    toast.innerHTML = `
+      <div class="toast-header">
+        <img class="baheth-logo" src="${getExtensionURL(prefersDarkMode ? "baheth-logo-light.svg" : "baheth-logo-dark.svg")}" alt="Baheth" />
+        <div class="loading-indicator"></div>
+      </div>
+      <div class="toast-content">
+        <p class="toast-title">جاري التحقق...</p>
+        <p class="toast-description">${
+          type === "video" ? "جاري التحقق من وجود المقطع على باحث" : "جاري التحقق من وجود قائمة التشغيل على باحث"
```

Comment:
unnecessary. showing a toast every time the user opens a video/playlist is annoying.